uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/OneAuthTestApp.java

```diff
@@ -0,0 +1,199 @@
+        UiAutomatorUtils.handleButtonClick("com.microsoft.oneauth.testapp:id/get_access_token_button");
+        try {
+            // Add a delay so that token can be retrived successfully
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        return getTokeSecret();
+    }
+
+    private void signIn(@NonNull final String username,
+                        @NonNull final String password,
+                        @NonNull final FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+        Logger.i(TAG, "Adding username and password on login screen..");
+
+        try {
+            final UiObject emailField = UiAutomatorUtils.obtainUiObjectWithTextAndClassType(
+                    "", EditText.class);
+            emailField.setText(username);
+            final UiObject nextBtn = UiAutomatorUtils.obtainUiObjectWithTextAndClassType(
+                    "Next", Button.class);
+            nextBtn.click();
+        } catch (UiObjectNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+        Logger.i(TAG, "Handle AAD Login page prompt..");
+        // handle prompt in AAD login page
+        new AadPromptHandler(promptHandlerParameters).handlePrompt(username, password);
+    }
+
+    public void handleBackButton() {
+        UiAutomatorUtils.pressBack();
+    }
+
+    @Override
+    public void addAnotherAccount(String username, String password, FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+
+    }
+
+    @Override
+    public void onAccountAdded() {
+
+    }
+
+    @Override
+    public void confirmAccount(@NonNull String username) {
+        // Make sure we are seeing the output text view
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+        try {
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+        } catch (UiObjectNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void assertSuccess() {
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+        final UiObject accountIdObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtAccountId");
+        try {
+            resultUIObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+            accountIdObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+            Assert.assertFalse(TextUtils.isEmpty(resultUIObject.getText()));
+            Assert.assertFalse(TextUtils.isEmpty(accountIdObject.getText()));
+        } catch (UiObjectNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void getAllAccounts() {
```

Comment:
Getting internal elements is still in progress.

Reply:
2023-06-09 15:11:28.942 30216-30250 UiObject                com....identity.client.sample.local  D  getText() = Mohit Singla
2023-06-09 15:11:28.942 30216-30250 Mohit                   com....identity.client.sample.local  D  **Inside if::Mohit Singla**
2023-06-09 15:11:28.944 30216-30250 QueryController         com....identity.client.sample.local  I  Matched selector: UiSelector[CHILD=UiSelector[INDEX=0, CLICKABLE=true], RESOURCE_ID=com.microsoft.oneauth.testapp:id/all_accounts_list] <<==>> [android.view.accessibility.AccessibilityNodeInfo@153e6; boundsInParent: Rect(0, 0 - 996, 126); boundsInScreen: Rect(42, 172 - 1038, 298); packageName: com.microsoft.oneauth.testapp; className: android.widget.TextView; text: Mohit Singla; error: null; maxTextLength: -1; contentDescription: null; tooltipText: null; viewIdResName: android:id/text1; checkable: false; checked: false; focusable: false; focused: false; selected: false; clickable: true; longClickable: false; contextClickable: false; enabled: true; password: false; scrollable: false; importantForAccessibility: true; visible: true; actions: [AccessibilityAction: ACTION_SELECT - null, AccessibilityAction: ACTION_CLEAR_SELECTION - null, AccessibilityAction: ACTION_CLICK - null, AccessibilityAction: ACTION_ACCESSIBILITY_FOCUS - null, AccessibilityAction: ACTION_NEXT_AT_MOVEMENT_GRANULARITY - null, AccessibilityAction: ACTION_PREVIOUS_AT_MOVEMENT_GRANULARITY - null, AccessibilityAction: ACTION_SET_SELECTION - null, AccessibilityAction: ACTION_SHOW_ON_SCREEN - null]]
2023-06-09 15:11:28.944 30216-30250 UiObject                com....identity.client.sample.local  D  getText() = Mohit Singla
2023-06-09 15:11:28.946 30216-30250 Mohit                   com....identity.client.sample.local  D  I**nside if::Cloud IDLAB Basic User 1**
**2023-06-09 15:11:28.949 30216-30250 Mohit                   com....identity.client.sample.local  D  [Mohit Singla, Cloud IDLAB Basic User 1]**

Printing accounts list